Solution1/WeatherApi/Startup.cs

```diff
@@ -23,13 +28,24 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            var connectionString = Configuration["ConnectionStrings:DefaultConnection"];
+            services.AddDbContext<RepositoryContext>(options => options.UseSqlServer(connectionString), ServiceLifetime.Singleton);
             services.Configure<AppConfiguration>(Configuration.GetSection(nameof(AppConfiguration)));
+            services.Configure<BackgroundJobConfiguration>(Configuration.GetSection(nameof(BackgroundJobConfiguration)));
+            services.Configure<WeatherApiConfiguration>(Configuration.GetSection(nameof(WeatherApiConfiguration)));
             services.Configure<RouteOptions>(options => options.LowercaseUrls = true);
+            
+            services.AddHangfire((provider, config) => config
+                .UseSqlServerStorage(connectionString)
+                .UseFilter(new ExceptionHangfireFilter(provider.GetService<ILogger<BackgroundJob>>())));
```

Comment:
don't like this new, rewrite it, now you pass dependencies by yourself, it should manage DI